test/e2e/framework/pod/wait.go

```diff
@@ -435,6 +435,19 @@ func PodsResponding(c clientset.Interface, ns, name string, wantName bool, pods
 	return wait.PollImmediate(poll, podRespondingTimeout, NewProxyResponseChecker(c, ns, label, name, wantName, pods).CheckAllResponses)
 }
 
+// WaitForNumberOfPods waits up to timeout to ensure that num pods in namespace ns are running.
+// Return the list of matching pods.
```

Comment:
We don't verify if the Pods are running actually:

```suggestion
// WaitForNumberOfPods waits up to timeout to ensure there are exact
// `num` pods in namespace `ns`.
// It returns the matching Pods or a timeout error.
```